olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/ApiResponse.h>
+#include <olp/core/client/CancellationToken.h>
+#include <olp/core/client/HRN.h>
+#include <olp/core/client/OlpClientSettings.h>
+#include <olp/dataservice/read/DataRequest.h>
+#include <olp/dataservice/read/DataServiceReadApi.h>
+#include <olp/dataservice/read/model/Data.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+using DataResult = model::Data;
```

Comment:
This will create a little bit of conflict if users includes both VersionedLayerClient and CatalogClient.
Either you include CatalogClient header here or move these to the VersionedLayerClient  class (i would not recommend this, as it will blow up when we add more and more APIs)

Reply:
I will move the callback definition right to the GetData method then